contracts/DisputableDelay.sol

```diff
@@ -0,0 +1,260 @@
+    * @param _delayedScriptId The ID of the script execution to challenge
+    */
+    function _onDisputableActionChallenged(uint256 _delayedScriptId, uint256 /* _challengeId */, address /* _challenger */)
+        internal
+    {
+        DelayedScript storage delayedScript = delayedScripts[_delayedScriptId];
+        delayedScript.pausedAt = getTimestamp64();
+        delayedScript.delayedScriptStatus = DelayedScriptStatus.Paused;
+
+        emit ExecutionPaused(_delayedScriptId, delayedScript.actionId);
+    }
+
+    /**
+    * @dev Allow script execution, IDisputable interface conformance
+    *      Note that either this, _onDisputableActionRejected() or _onDisputableActionVoided()
+    *      will be called once for each challenge
+    * @param _delayedScriptId The ID of the script execution to allow
+    */
+    function _onDisputableActionAllowed(uint256 _delayedScriptId) internal {
+        DelayedScript storage delayedScript = delayedScripts[_delayedScriptId];
+        uint64 timePaused = getTimestamp64().sub(delayedScript.pausedAt);
+
+        delayedScript.executionFromTime = delayedScript.executionFromTime.add(timePaused);
+        delayedScript.pausedAt = 0;
+        delayedScript.delayedScriptStatus = DelayedScriptStatus.Active;
+
+        emit ExecutionResumed(_delayedScriptId, delayedScript.actionId);
+    }
+
+    /**
+    * @dev Reject script execution, IDisputable interface conformance
+    *      Note that either this, _onDisputableActionAllowed() or _onDisputableActionVoided()
+    *      will be called once for each challenge
+    * @param _delayedScriptId The ID of the script execution to reject
+    */
+    function _onDisputableActionRejected(uint256 _delayedScriptId) internal {
+        DelayedScript storage delayedScript = delayedScripts[_delayedScriptId];
+
+        delayedScript.delayedScriptStatus = DelayedScriptStatus.Cancelled;
+
+        emit ExecutionCancelled(_delayedScriptId, delayedScript.actionId);
+    }
+
+    /**
+    * @dev Void script execution, IDisputable interface conformance
+    *      Note that either this, _onDisputableActionAllowed() or _onDisputableActionRejected()
+    *      will be called once for each challenge
+    * @param _delayedScriptId The ID of the script execution to void
+    */
+    function _onDisputableActionVoided(uint256 _delayedScriptId) internal {
+        _onDisputableActionAllowed(_delayedScriptId);
+    }
+
+    /**
+    * @notice Execute the script with ID `_delayedScriptId`
+    * @param _delayedScriptId The ID of the script to execute
+    */
+    function execute(uint256 _delayedScriptId) external nonReentrant {
+        DelayedScript storage delayedScript = delayedScripts[_delayedScriptId];
+        require(_canExecute(delayedScript), ERROR_CANNOT_EXECUTE);
+
+        (,,,,,bool closed,,) = _ensureAgreement().getAction(delayedScript.actionId);
+        if (!closed) {
+            _closeAgreementAction(delayedScript.actionId);
+        }
```

Comment:
This check will be implemented either in the base app or the agreement app directly. Although it is not implemented yet, we can probably remove it for now:

```suggestion
        _closeAgreementAction(delayedScript.actionId);
```